backend/controllers/userController.js

```diff
@@ -87,4 +91,226 @@ const loadUser = catchAsyncError(async (req, res, next) => {
   });
 });
 
-export { signupUser, loginUser, logoutUser, loadUser };
+//Forgot Password Controller
+const forgotPassword = catchAsyncError(async (req, res, next) => {
+  const { email } = req.body;
+  const user = await User.findOne({ email: email });
+  if (!user) return next(new ErrorHandler("User doesn't exist", 401));
+
+  //Generating Reset Password Token
+  const resetToken = await user.getResetPasswordToken();
+
+  //Saving Token to Database
+  await user.save();
+
+  //Reset Password Url
+  const resetPasswordUrl = `${req.protocol}://${req.hostname}:${process.env.FRONTEND_PORT}/reset-password/${resetToken}`;
+  const resetPasswordUrlM = `${req.protocol}://192.168.1.6:${process.env.FRONTEND_PORT}/reset-password/${resetToken}`;
+
+  console.log(resetPasswordUrl); //Just to get Reset Url directly from Console
+
+  const mailDetails = {
+    to: user.email,
+    subject: "NoteTake - Password Reset",
+    // text: `Hi, ${user.name} \nPlease click below to reset your password for NoteTake. If you did not request a new password, please ignore this email.\n
+    // ${resetPasswordUrl}`,
+    html: `
```

Comment:
You should write functions for different types of Emails, that will improve code quality and readability

Reply:
Please check I have made changes

- fix long html in mail options object
- by making different functions in emailTemplates for each type of email
- as per reviewers comment